src/editors.ts

```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+import * as vs from 'vscode';
+
+export function hasActiveDartEditor(): boolean {
+  if (!vs.window.activeTextEditor)
+  	return false;
+
+  return vs.window.activeTextEditor.document.fileName.endsWith(".dart");
```

Comment:
In other places we use the `languageId` on the `editor`(`document`?). This means if the user changes the language for a document (done in the status bar) it'll honour their language (though whether it's likely someone will change a `.dart` file to `HTML` and then expect the Dart things to vanish and HTML ones to appear, I don't know!)